scripts/oneoff/migrate-supplier-organisation-size-from-declarations.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+"""Selectively migrate supplier organisation size from a recent framework declaration to the field on supplier itself.
+
+   Based on scripts/oneoff/migrate-supplier-data-from-declarations.py (retry/backoff explained there in more detail)
+
+Usage:
+    scripts/oneoff/migrate-supplier-organisation-size-from-declarations.py <stage> <data_api_token>
+        [<user>] [--dry-run]
+
+Positional arguments:
+    <stage>                                 API stage to perform operation on
+    <data_api_token>                        appropriate API token for <stage>
+
+Optional arguments:
+    -h, --help                              show this help message and exit
+    <user>                                  Audit Trail 'updated_by' user
+    --dry-run                               skip update step
+"""
+import sys
+sys.path.insert(0, '.')
+
+from datetime import datetime
+import logging
+import getpass
+
+import backoff
+from docopt import docopt
+import requests
+from dmapiclient import DataAPIClient, HTTPError
+from dmapiclient.errors import HTTPTemporaryError
+
+from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage
+from dmscripts.helpers.logging_helpers import configure_logger
+from dmscripts.helpers.logging_helpers import INFO as loglevel_INFO
+
+
+logger = logging.getLogger("script")
+
+
+_backoff_wrap = backoff.on_exception(
+    backoff.expo,
+    (HTTPTemporaryError, requests.exceptions.ConnectionError, RuntimeError),
+    max_tries=5,
+)
+
+
+def _catch_404_none(func):
+    try:
+        return func()
+    except HTTPError as e:
+        if e.status_code == 404:
+            return None
+        else:
+            raise
+
+
+def _get_supplier_frameworks(data_api_client, supplier_id):
+    # Get all supplierFrameworks for this supplier that have a declaration
+    try:
+        supplier_frameworks = _catch_404_none(
+            _backoff_wrap(
+                lambda: data_api_client.get_supplier_frameworks(supplier_id)
```

Comment:
Yeah, @risicle might tell you to use `functools.partial`, but eh :-)

Reply:
All that stuff is copied from the the other script 😸

Reply:
Oh well in that case maybe it should be in `dmscripts` :-)